.github/workflows/build-and-test.yml

```diff
@@ -45,7 +41,11 @@ jobs:
         with:
           java-version: '17'
           distribution: 'temurin'
-          cache: 'gradle'
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
```

Comment:
I'm shifting to [gradle-build-action](https://github.com/gradle/gradle-build-action) as this looks to be a better way to bootstrap gradle in GHA. Of note, it automatically handles cache restore, etc. which is nice.